core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3221,31 +3221,35 @@ protected function save($varValue)
 		// Save the value if there was no error
 		if (((string) $varValue !== '' || !($arrData['eval']['doNotSaveEmpty'] ?? null)) && ($this->varValue !== $varValue || ($arrData['eval']['alwaysSave'] ?? null)))
 		{
+			$varEmpty = Widget::getEmptyValueByFieldType($GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql'] ?? array());
+			$strType = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql']['type'] ?? null;
```

Comment:
Should we apply the `array_filter()` here already instead of three times in the code below?

Reply:
I thought about this as well, but I think this is more explicit. The types array is related to the parameters given to the SQL method. This **string** is just the type of the current field.

Reply:
Yeah, but `$strType` is not used anywhere else, so why declare the variable at all?

Reply:
not sure what you mean. The array is created "on the fly" to match the other argument in the specific place. How else would you get `$strType`?

Maybe to make this more clear: The values and types passed to `->query()` **must** have the same array keys to work. Reading line 3248 to 3252 is all I need to understand that this is correct. If we build the array before, one has to analyse the whole method to mentally make sure that's correct. I even had to look up in the manual what array keys `array_unshift` produces for the types to work correctly 🙃

Reply:
Not sure about that. There are some basic programming principles and one of it is DRY – don't repeat yourself. And right now, you are repeating yourself, because your are executing `array_filter(array($strType)` three times in the same method instead of just once. 🤷‍♂️

Reply:
https://softwareengineering.stackexchange.com/a/194507 🤷 
ultimately I don't care, feel free to CS it if you have to.

Reply:
```suggestion
			$arrTypes = array_filter(array($GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql']['type'] ?? null));
```

Reply:
See cac1c988a72ebeb13f3107d8fc1b996fcf27ab1d